src/plugins/credential_management/server/credential/credential_manager.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import { OpenSearchDashboardsRequest } from 'opensearch-dashboards/server';
+import { CryptoCli } from '../crypto';
+
+interface Credential {
+    readonly credential_name: string,
+    readonly credential_type: string,
+    readonly credential_material: BasicAuthCredentialMaterial | AWSIAMCredentialMaterial,
+}
+
+interface BasicAuthCredentialMaterial{
+    readonly user_name: string
+    readonly password: string
+}
+
+interface AWSIAMCredentialMaterial {
+    readonly encrypted_aws_iam_credential: string
+}
+
+// TODO: Refactor handler, add logger, etc
+export async function createHandler(request: OpenSearchDashboardsRequest) {
+    const cryptoCli = CryptoCli.getInstance()
+    if (request.body.credential_type == 'basic_auth') {
```

Comment:
a good opportunity to use factory pattern to increase extensibility and testability

const credential = credentialFactory.create(options)

not required, think about it.


https://blog.sessionstack.com/how-javascript-works-the-factory-design-pattern-4-use-cases-7b9f0d22151d